src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -257,13 +257,15 @@ private static string GetIdentifierFromPath(string filePath)
             return builder.ToString();
         }
 
-        private static ParallelOptions GetParallelOptions()
+        private static ParallelOptions GetParallelOptions(GeneratorExecutionContext generatorExecutionContext)
         {
             if (Debugger.IsAttached)
             {
                 return new ParallelOptions { MaxDegreeOfParallelism = 1 };
             }
-            return DefaultParallelOptions;
+            var options =  DefaultParallelOptions;
+            options.CancellationToken = generatorExecutionContext.CancellationToken;
+            return options;
```

Comment:
We should use a non-static instance if we're mutating this:

```suggestion
            var options = new ParallelOptions { CancellationToken = generatorExecutionContext.CancellationToken };
```